taskipy/pyproject.py

```diff
@@ -0,0 +1,65 @@
+from pathlib import Path
+from typing import Any, MutableMapping
+
+import toml
+
+from taskipy.exceptions import (
+    MalformedPyProjectError,
+    MissingPyProjectFileError,
+    MissingTaskipyTasksSectionError
+)
+
+
+class PyProject:
+    def __init__(self, file_name: str = "pyproject.toml", directory: Path = Path.cwd()):
+        """
+        Initializes PyProject by setting up the path to and loading in
+        the pyproject.toml file in the current directory.
+
+        Args:
+            file_name (str, optional): The name of the pyproject file. Defaults to 'pyproject.toml'.
+            directory (Path, optional): The directory that the pyproject file is in. Defaults to Path.cwd().
+        """
+        self.path = Path(directory / file_name).resolve()
+        self.items = self.__load_toml_file()
+
+    @property
+    def tasks(self) -> dict:
+        """Retrieve the taskipy tasks.
+
+        Raises:
+            MissingTaskipyTasksSectionError: If there is a missing taskipy tasks section.
+
+        Returns:
+            dict: The taskipy tasks.
+        """
+        try:
+            return self.items["tool"]["taskipy"]["tasks"]
+        except KeyError:
+            raise MissingTaskipyTasksSectionError()
+
+    @property
+    def settings(self) -> dict:
```

Comment:
let's do the same thing as in the `tasks` property: throw a unique `MissingTaskipySettingsSectionError` instead of relying on `KeyError` here (as we do in `task.py`)